### PR TITLE
Infrastructure for on demand key-exchange

### PR DESCRIPTION
**Pull request by yontyon**
In this PR we introduce a basic (and working) infrastructure for on-demand key exchange.
We add a key exchange command and handlers to the reconfiguration handler.
In addition, we have a small apollo test that manages to send the key exchange message

### CODE REVIEW

**Comment by andrewjstone** [comment]
@yontyon Can you please create a commit message with more information about what `KE` is and what the goal of this PR is? Please read https://chris.beams.io/posts/git-commit/ for details on what is expected.

**Comment by andrewjstone** [comment]
Thanks for the excellent commit message @yontyon

File: kvbc/include/kvbc_key_types.hpp [hunk 1]
```diff
@@ -18,4 +18,5 @@ static const char pruning_last_agreed_prunable_block_id_key = 0x24;
 static const char reconfiguration_wedge_key = 0x25;
 static const char reconfiguration_download_key = 0x26;
 static const char reconfiguration_install_key = 0x27;
+static const char reconfiguration_key_exchange = 0x27;
```
Comment (robem) [inline kvbc/include/kvbc_key_types.hpp:21]: Why is it the same as install key?

Comment (yontyon) [reply kvbc/include/kvbc_key_types.hpp]: good catch. Thanks

**Review by toly-kournik — APPROVED** [review]
LGTM!

**Pull request merged**